src/vorta/views/main_window.py

```diff
@@ -277,27 +277,26 @@ def closeEvent(self, event):
             SettingsModel.key == 'previous_window_height'
         ).execute()
 
-        if not is_system_tray_available():
-            if SettingsModel.get(key="enable_background_question").value:
-                msg = QMessageBox()
-                msg.setStandardButtons(QMessageBox.Yes | QMessageBox.No)
-                msg.setParent(self, QtCore.Qt.Sheet)
-                msg.setText(self.tr("Should Vorta continue to run in the background?"))
-                msg.button(QMessageBox.Yes).clicked.connect(
-                    lambda: self.miscTab.save_setting("disable_background_state", True)
-                )
-                msg.button(QMessageBox.No).clicked.connect(
-                    lambda: (
-                        self.miscTab.save_setting("disable_background_state", False),
-                        self.app.quit(),
-                    )
+        if SettingsModel.get(key="enable_background_question").value:
+            msg = QMessageBox()
+            msg.setStandardButtons(QMessageBox.Yes | QMessageBox.No)
+            msg.setParent(self, QtCore.Qt.Sheet)
+            msg.setText(self.tr("Should Vorta continue to run in the background?"))
+            msg.button(QMessageBox.Yes).clicked.connect(
+                lambda: self.miscTab.save_setting("disable_background_state", True)
+            )
+            msg.button(QMessageBox.No).clicked.connect(
+                lambda: (
+                    self.miscTab.save_setting("disable_background_state", False),
+                    self.app.quit(),
                 )
-                msg.setWindowTitle(self.tr("Quit"))
-                dont_show_box = QCheckBox(self.tr("Don't show this again"))
-                dont_show_box.clicked.connect(lambda x: self.miscTab.save_setting("enable_background_question", not x))
-                dont_show_box.setTristate(False)
-                msg.setCheckBox(dont_show_box)
-                msg.exec()
-            elif not SettingsModel.get(key="disable_background_state").value:
-                self.app.quit()
+            )
+            msg.setWindowTitle(self.tr("Quit"))
+            dont_show_box = QCheckBox(self.tr("Don't show this again"))
+            dont_show_box.clicked.connect(lambda x: self.miscTab.save_setting("enable_background_question", not x))
```

Comment:
Minor nit: we recently added a flake8 rule against single-letter variables in Borgmatic; maybe Vorta could benefit from it too.
https://github.com/borgmatic-collective/borgmatic/commit/f42890430c59a40a17d9a68a193d6a09674770cb#diff-43f6847f9f058bf2ba296a9a3554506628b2dc79c663d4f7c266ba5589d7d618R10

Reply:
I find single letter vars fine in some situations. Like a loop or for maths.

Reply:
> Suggestion: we recently added a flake8 rule against single-letter variables in Borgmatic; maybe Vorta could benefit from it too. [borgmatic-collective/borgmatic@f428904#diff-43f6847f9f058bf2ba296a9a3554506628b2dc79c663d4f7c266ba5589d7d618R10](https://github.com/borgmatic-collective/borgmatic/commit/f42890430c59a40a17d9a68a193d6a09674770cb#diff-43f6847f9f058bf2ba296a9a3554506628b2dc79c663d4f7c266ba5589d7d618R10)

Thanks! will look into that 😄

Reply:
You don't have to, at least not in this PR. It's okay for vorta's case i guess.

Reply:
Yeah, this line of code doesn't need any change.